### PR TITLE
plots: support opening plots file directly in the browser

### PR DESCRIPTION
**Pull request by skshetry**
Closes https://github.com/iterative/dvc/issues/4179
```
$ dvc plots show --open
```
I don't know if this is that useful, but I think this does save some time (I was feeling the need when I was going through `example-get-started`).

### Alternative:
No way to be compatible across different platforms. For me on Linux, it's following:
```
$ dvc plots show | xargs xdg-open
```

Working on macOS and on Windows (cc @jorgeorpinel?) confirmation would be great. 

* [x] ❗ I have followed the [Contributing to DVC](https://dvc.org/doc/user-guide/contributing/core) checklist.

* [x] 📖 If this PR requires [documentation](https://dvc.org/doc) updates, I have created a separate PR (or issue, at least) in [dvc.org](https://github.com/iterative/dvc.org) and linked it here.

Thank you for the contribution - we'll try to review it as soon as possible. 🙏

### CODE REVIEW

**Comment by shcheklein** [comment]
My workflow for this is to use `Cmd (~Ctrl on Mac) + click the link DVC prints` - it opens it automatically in the browser. Not saying that this option is not needed, just sharing my thoughts on why this feature didn't cross my mind I guess. I wonder how many users are familiar with hot keys in their terminals.

File: tests/unit/command/test_plots.py [hunk 1]
```diff
@@ -103,3 +105,39 @@ def test_plots_show_vega(dvc, mocker, caplog):
...
+    assert cmd.run() == 1
+
+    expected_url = f"file://{tmp_dir / 'plots.html'}"
+    mocked_open.assert_called_once_with(expected_url)
+
+    error_message = "Failed to open. Please try opening it manually."
+    assert caplog.record_tuples == [
+        ("dvc.command.plots", logging.INFO, expected_url),
+        ("dvc.command.plots", logging.ERROR, error_message),
+    ]
```
Comment (pared) [inline tests/unit/command/test_plots.py:143]: I was thinking about adding `--open` option to `test_plots_diff`, this exceeds expectations!

**Review by pared — APPROVED** [review]

**Comment by skshetry** [comment]
> @skshetry Btw, looks like even though the checkbox is ticket, the doc is missing?

I was not sure if this was going to merged. I have created the docs, thanks.

**Comment by jorgeorpinel** [comment]
> Working on macOS and on Windows (cc @jorgeorpinel?) confirmation would be great.

I'm on Windows but on WSL (Ubuntu) and it fails there unfort (terminal-only Linux):

```console
$ dvc plots show --open train.json
file:///home/jop/DVC-repos/test/plots.html
Start : This command cannot be run due to the error: The system cannot find the file specified.
At line:1 char:1
+ Start "file:///home/.../plots.html"
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (:) [Start-Process], InvalidOperationException
    + FullyQualifiedErrorId : InvalidOperationException,Microsoft.PowerShell.Commands.StartProcessCommand
```

**Comment by jorgeorpinel** [comment]
Feel free to lmk when this get's to Windows installer so I can try on regular `cmd`.

**Comment by jorgeorpinel** [comment]
Ah jk. I have the .exe now... ⌛

**Comment by jorgeorpinel** [comment]
On Windows / Cmder (Git Bash) it works for me! But only from a DVC project dir.

Minor: for some reason if you run it outside a DVC project it get's stuck (while on Linux that seems to work regardless of where you are).

**Comment by jorgeorpinel** [comment]
Same on regular Command Prompt (`cmd`):

- `plots show` (in general) gets stuck if run outside a DVC project dir.
- Works from within a DVC project 🙂

**Comment by skshetry** [comment]
@jorgeorpinel, opened https://github.com/iterative/dvc/pull/5670 to address the WSL related issue. 

To open into a browser in the host computer, you have to specify `BROWSER` envvar, something like follows:
```
export BROWSER=/mnt/c/Program Files (x86)/Google/Chrome/Application/chrome.exe
```

and then you can do `--open` and it'll open in the specified browser. For me, it's terribly slow.

**Comment by skshetry** [comment]
@jorgeorpinel, for other issues, please open an issue as it needs more information on what's happening. Thanks. 🙂

**Pull request merged**